### ISSUE TITLE
facebook login works only at 2nd time.

### ISSUE THREAD

**Issue by RonAlmog** [angular-fullstack/generator-angular-fullstack#2360]
- [x] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 3.7.5
Node | 6.2
npm | 3.9.2
Operating System | Windows 10 
etc | etc

Item | Answer
----- | -----
Transpiler | Babel 
Markup | HTML
DB | MongoDB
Auth | N
etc | etc

i'm working on enabling the users to login using facebook auth.  currently this is what happens:
at the login page, when you hit 'connect with facebook'. it brings the facebook login page,
you enter your user/pass, and it takes you back to our login page as if nothing happened.
the server method was hit, i see data coming from facebook, and a record is created in 'users'.
but we are not logged in.
then you try the second time. now it works, and you are logged in with your facebook name.

what can it be? any directions to look?

**Comment by RonAlmog** [comment]
Additional info: It's the same for all 3 auth sources: fb, google, twitter.  
First time it takes you to the site (fb, google, etc), doing the validation, then takes your back to your site, not logged in. (user created in db).
You hit 'connect with facebook' again, and you're logged in right away.
Maybe that's how it's planned to be?

**Comment by atodicebear** [comment]
Maybe its the same Problem as here mentioned?
#2224 
Some Error/Warning outputs?
We will have to wait for Investigation/Solution

**Comment by RonAlmog** [comment]
maybe. but in my case i don'e see any CSRF Token missing message. 
what i can see is a call to http://localhost:9000/api/users/me  that returns with 401.
i did not change that function, it's pretty basic. just finding the user based on req.user._id;
when i inspect the call to /api/users/me, it looks ok to me:

```
Authorization:Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0Nzk5MTExMDUsImV4cCI6MTQ3OTkyOTEwNX0.0LzUIPo-0J9IlqI3s-TFaG9IUaXEXZe1CkyN1xAaBb8
Cache-Control:no-cache
Connection:keep-alive
Cookie:connect.sid=s%3A1wHvTY1hU2ir2xoMqK4lQutf8l2o-1A1.m2VAF0H8YEI99FZ0PQddbcULSeNwSHVXkB4w%2FBxO8MM; _gat=1; TawkConnectionTime=0; __tawkuuid=e::localhost::YcnpEeeF5DNcZqctTCiIPrf8eRzOFLCer7U8v0zUfNBJPJGUe2FlRovOZs5QbQ1R::2; Tawk_57fda888277fb7280d9808ac=vs56.tawk.to:443::0; token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0Nzk5MTExMDUsImV4cCI6MTQ3OTkyOTEwNX0.0LzUIPo-0J9IlqI3s-TFaG9IUaXEXZe1CkyN1xAaBb8; _ga=GA1.1.1633099374.1475952063; XSRF-TOKEN=SKsAfxkbYaQKM2Ds8v7Lac6%2BGsSXO9zbenAdw%3D
Host:localhost:9000
```
so i can't tell exactly why it's failing...  
any tip would be greatly appreciated.

**Comment by taotau** [comment]
I havent played with the newer versions of this generator, but this sounds like it could be a similar problem to what I describe in #1450 ?

**Comment by RonAlmog** [comment]
YES! that did the trick! wow, thanks so much for this! i have spent so many hours on this...
now it works great on first attempt. thanks again!

so let me repeat, for anybody who's looking here in the future:
in passport.js (both fb, google, all others), go down to where the new user is saved (user.saveAsync),
and change it to:
```
user.saveAsync()
            .then(function(user) {
              return done(null, user[0]);
            })
```

**Comment by Awk34** [comment]
@RonAlmog is this issue only with the 3.x.x versions?

We'll probably want to patch this

**Comment by RonAlmog** [comment]
tested only in 3.7.5 (both problem and solution). i don't know about 4.1

**Comment by Abdizriel** [comment]
@Awk34 in 4.x.x there's the same issue
